mobile-app/app/screens/AppNavigator/screens/Loans/hooks/DFIRequirementForDusdLoanAndCollateral.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  CollateralToken,
+  LoanToken,
+  LoanVaultTokenAmount,
+} from "@defichain/whale-api-client/dist/api/loan";
+import { RootState } from "@store";
+import BigNumber from "bignumber.js";
+import { useSelector } from "react-redux";
+import { getActivePrice } from "../../Auctions/helpers/ActivePrice";
+
+interface useDFIRequirementForDusdLoanAndCollateralProps {
+  collateralAmounts: LoanVaultTokenAmount[];
+  loanAmounts: LoanVaultTokenAmount[];
+  collateralValue: BigNumber;
+  loanValue: BigNumber;
+  loanToken: LoanToken;
+  minColRatio: string;
+}
+
+/**
+ * To check if current vault has at least 50% of DFI as total collateral when taking DUSD loan
+ *
+ * Modified formula from [DeFiCh/ain] to include new loan amount to be taken during `Borrow` flow
+ *
+ * Source: https://github.com/DeFiCh/ain/blob/a2a8ee7c12649319456b247b52164cba0727f7db/src/masternodes/mn_checks.cpp#L3003
+ *
+ * @returns
+ */
+export function useDFIRequirementForDusdLoanAndCollateral(
+  props: useDFIRequirementForDusdLoanAndCollateralProps
+) {
+  const collateralTokens: CollateralToken[] = useSelector(
+    (state: RootState) => state.loans.collateralTokens
+  );
+  const hasDUSDLoan = props.loanAmounts.some(
+    (loan) => loan.displaySymbol === "DUSD"
+  );
+  const isTakingDUSDLoan = props.loanToken.token.displaySymbol === "DUSD";
+
+  const dfiCollateralToken = collateralTokens.find(
+    (col) => col.token.displaySymbol === "DFI"
+  );
+  const dfiActivePrice = getActivePrice(
+    "DFI",
+    dfiCollateralToken?.activePrice,
+    dfiCollateralToken?.factor,
+    "ACTIVE",
+    "COLLATERAL"
+  );
+  const dfiCollateralValue = new BigNumber(dfiActivePrice).multipliedBy(
+    props.collateralAmounts.find((col) => col.displaySymbol === "DFI")
+      ?.amount ?? 0
+  );
+  const isDFILessThanHalfOfRequiredCollateral = dfiCollateralValue.isLessThan(
+    new BigNumber(props.loanValue)
```

Comment:
I will also add this same logic for Remove collateral, so just want to align:
1. This rule should only apply if vault has DUSD collateral (and borrowing DUSD)
2. As long as DUSD `loanValue` and collateral is greater than zero, 50%DFI is required. User shouldn't be able to take DUSD loan at all

This is based on this [this article](https://blog.defichain.com/new-rules-guide-decentralized-loans/) 🤔

Reply:
1. For a vault that has taken any loan, it will have a certain amount of DFI and/or DUSD as collateral, because that is the basic requirement to take loan. Any other collateral added will not increase amount of loan you can take if 50% of the total collateral to satisfy min col ratio is not DFI and/or DUSD
2. The new rule is actually trying to limit DUSD as being the majority of the collateral IF the vault is trying to take DUSD loan. Instead of allowing any ratio of DFI and DUSD as collateral, we now force it to have the original requirement on my previous point to only include DFI to satisfy the min col ratio.

tldr: the reason i didnt include checking for DUSD collateral because checking for DFI collateral is enough to decide